agent/agent.go

```diff
@@ -1921,17 +1926,25 @@ OUTER:
 			}
 
 			for segment, coord := range cs {
+				agentToken := a.tokens.AgentToken()
 				req := structs.CoordinateUpdateRequest{
 					Datacenter:   a.config.Datacenter,
 					Node:         a.config.NodeName,
 					Segment:      segment,
 					Coord:        coord,
-					WriteRequest: structs.WriteRequest{Token: a.tokens.AgentToken()},
+					WriteRequest: structs.WriteRequest{Token: agentToken},
 				}
 				var reply struct{}
+				//  todo(mkcp) port all of these logger calls to hclog w/ loglevel configuration
 				if err := a.RPC("Coordinate.Update", &req, &reply); err != nil {
 					if acl.IsErrPermissionDenied(err) {
-						a.logger.Printf("[WARN] agent: Coordinate update blocked by ACLs")
+						_, tokenIdent, err2 := a.resolveIdentityFromToken(agentToken)
+						if err2 != nil {
```

Comment:
Should we be handling this kind of err in the call site, or should this be handled in another function. I feel like the error case here is not particularly important to the critical path of this code (it's a log, it either logs or it doesn't) and wonder if this ought to be abstracted somewhere.

Reply:
I agree, I'll play around w/ lifting up that error handling